modules/common/util/template_util.go

```diff
@@ -57,6 +57,48 @@ type Template struct {
 	ConfigOptions      map[string]interface{} // map of parameters as input data to render the templates
 	SkipSetOwner       bool                   // skip setting ownership on the associated configmap
 	Version            string                 // optional version string to separate templates inside the InstanceType/Type directory. E.g. placementapi/config/18.0
+	PostProcessCleanup bool                   // optional boolean parameter to remove extra new lines from service confs, by default set to false
```

Comment:
I still prefer passing in the function references to run as post processing as that allows for a lot more flexibility in the future. 

But if the others wants a separate boolean for each future post processing then at least be super specific what this boolean enables. The code Sean provided not just remove extra empty lines from the rendered template but also insert missing empty lines between ini like sections.

Reply:
We had a call with @auniyal61 and I learned that a single Template instance can actually mean a list of rendered files. Both via AdditionalTemplate and the logic that picks up a set of files from a path + glob. This means that a single PostProcessCleanup bool is actually applied not to a single file but to a set of diverse files(potentially a mix of different file types like ini, kolla json, apache conf). This would also be true for the set of functions as well if we change this to that type. This makes the situation more complicated as post processing cannot be configured per template file. Changing the Template to allow per file post processing is seems to be a big work so I drop the idea out of the scope of this change.

@stuggi How do you feel about eventually refactoring Template? I see three possible directions we could go
1. keep a single Template struct to pick up multiple files and render them as today, and then extend the interface to make such rendering still configurable per template file level
2. break the coupling between template rendering and Secret / CM creation with the rendered template. Make it two separate, client callable, steps. Like below to allow post processing on the client side.
   ```
   renderedTemplates = utils.RenderTemplate(template utils.Template)
   // the service operator can do post processing here on the client side
   utils.CreateCM/Secret(renderedTemplates)
   ```
3. break the current Template struct to have a list of Template structs one for each template file making the template rendering configuration per template file struct.

Reply:
Hi @stuggi , can you please suggest on this.

Reply:
@auniyal61 sorry for the delay. I was out last week. I'll get back to this asap.

Reply:
@gibizer @SeanMooney soffy for the delay, I finally had some time today to look into this. What you think about this approach [1]. This adds new template functions , especially for our use case here `execTempl` and `removeNewLines`. With this you can define your template inline and then render the template using the `execTempl` and remove multiple empty lines using `removeNewLines`. An example on how you could use that in a template is shown in the description.

Specific for the nova.conf in this example PR [2]. The benefit is that you could control this just from inside the template and no other code change is required.

[1] https://github.com/openstack-k8s-operators/lib-common/pull/569
[2] https://github.com/openstack-k8s-operators/nova-operator/pull/874